corehq/util/pagination.py

```diff
@@ -182,6 +195,31 @@ def _get_event_handler(self):
         else:
             return ResumableIteratorEventHandler(self)
 
+    def retry(self, item_id, max_retry=3):
+        """Add document to be yielded at end of iteration
+
+        Iteration order of retry documents is undefined. All retry
+        documents will be yielded after the initial non-retry phase of
+        iteration has completed, and every retry document will be
+        yielded each time the iterator is stopped and resumed during the
+        retry phase. This method is relatively inefficient because it
+        forces the iteration state to be saved to couch. If you find
```

Comment:
Thanks for your perspective Ben. I hadn't thought about a dashboard but
that would be nice.

I certainly don't plan on changing this behaviour. If the same iteration
needs to be run on multiple databases then the caller can ensure that the
iteration key is unique for the different DBs.